handlers/handlers.go

```diff
@@ -0,0 +1,360 @@
+	if repeat == "" {
+		return "", errors.New("repeate is empty")
+
+	}
+	if strings.Contains(repeat, "d ") {
+		daysToAdd, err := strconv.Atoi(strings.TrimPrefix(repeat, "d "))
+		if err != nil {
+			return "", err
+		}
+		if daysToAdd > 400 {
+			return "", errors.New("repeat period in days max 400")
+		}
+
+		beginDate, err := time.Parse(DateFormat, date)
+		if err != nil {
+			return "", err
+		}
+
+		newDate := beginDate.AddDate(0, 0, daysToAdd)
+		for newDate.Before(now) {
+			newDate = newDate.AddDate(0, 0, daysToAdd)
+		}
+
+		return newDate.Format(DateFormat), nil
+
+	}
+	if repeat == "y" {
+		beginDate, err := time.Parse(DateFormat, date)
+		if err != nil {
+			return "", err
+		}
+
+		newDate := beginDate.AddDate(1, 0, 0)
+
+		for newDate.Before(now) {
+			newDate = newDate.AddDate(1, 0, 0)
+		}
+
+		return fmt.Sprint(newDate.Format(DateFormat)), nil
+	} else {
+		return "", errors.New("repeat string has wrong format")
+	}
+}
+
+func NextDateGET(w http.ResponseWriter, r *http.Request) {
+	pNow, err := time.Parse(DateFormat, r.FormValue("now"))
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	pDate := r.FormValue("date")
+	pRepeat := r.FormValue("repeat")
+	newDate, err := nextDate(pNow, pDate, pRepeat)
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+	_, err = w.Write([]byte(newDate))
+
+	if err != nil {
+		http.Error(w, fmt.Sprintf("nextDateGET error: %w", err), http.StatusBadRequest)
+	}
+}
+
+func responseWithError(w http.ResponseWriter, err error) {
+	log.Printf("Error: %w", err)
+
+	error, _ := json.Marshal(models.ResponseError{Error: err.Error()})
+
+	w.Header().Set("Content-Type", "application/json")
+	_, err = w.Write(error)
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+	}
+}
+
+func TaskPost(w http.ResponseWriter, r *http.Request) {
+	var task models.Task
+	var buf bytes.Buffer
+
+	if _, err := buf.ReadFrom(r.Body); err != nil {
+		responseWithError(w, err)
+		return
+	}
+
+	if err := json.Unmarshal(buf.Bytes(), &task); err != nil {
+		responseWithError(w, err)
+		return
+	}
+
+	//проверки корректного заполнения полей
+	if len(task.Title) == 0 {
+		responseWithError(w, errors.New("title is empty"))
+		return
+	}
+
+	if len(task.Date) == 0 {
+		task.Date = time.Now().Format(DateFormat)
+	} else {
+		_, err := time.Parse(DateFormat, task.Date)
+		if err != nil {
+			responseWithError(w, errors.New("bad date"))
+			return
+		}
+
+		if len(task.Repeat) > 0 {
+			if nextDate, err := nextDate(time.Now(), task.Date, task.Repeat); err != nil {
+				responseWithError(w, err)
+				return
+			} else if task.Date < time.Now().Format(DateFormat) {
+				task.Date = nextDate
+			}
+		}
+
+		if task.Date < time.Now().Format(DateFormat) {
+			task.Date = time.Now().Format(DateFormat)
+		}
+	}
+
+	if result := database.Db.Create(&task); result.Error != nil {
+		log.Fatalf("Err: %s", result.Error)
```

Comment:
`Fatalf` аварийно завершает программ и все падает, тут нужно ту
```suggestion
		log.Errorf("Err: %s", result.Error)
```